src/script_interface/Exception.hpp

```
@@ -16,13 +16,22 @@
  * You should have received a copy of the GNU General Public License
  * along with this program.  If not, see <http://www.gnu.org/licenses/>.
  */
-#ifndef SCRIPT_INTERFACE_GLOBALS_HPP
-#define SCRIPT_INTERFACE_GLOBALS_HPP
+#ifndef ESPRESSO_EXCEPTIONS_HPP
+#define ESPRESSO_EXCEPTIONS_HPP
 
-#include "thermostat.hpp"
+#include <stdexcept>
+#include <string>
 
-extern LangevinThermostat langevin;
-extern BrownianThermostat brownian;
-extern IsotropicNptThermostat npt_iso;
+namespace ScriptInterface {
+struct Exception : public std::exception {
+  explicit Exception(const char *msg) : message(msg) {}
+  explicit Exception(std::string msg) : message(std::move(msg)) {}
```

What is the purpose of this custom exception class? It has a generic name, yet seems to be only used in `AutoParameters` for read-only and unknown parameters. Can it be safely used elsewhere in the script interface?

---

This is so that we can catch all exceptions from the script interface but not others.